### ISSUE TITLE
[uncontrolled] onBeforeChange's third parameter always null

### ISSUE THREAD

**Issue by euZebe** [scniro/react-codemirror2#61]
_value_ parameter seems to always be null in onBeforeChange(editor,_ data, value, next) for an uncontrolled codemirror ; did I miss something ?
```
import React, { Component } from 'react';
import { UnControlled as CodeMirror } from 'react-codemirror2';

class App extends Component {
  handleBeforeChange = (editor, data, value, next) => {
    console.log('handleBeforeChange', value); // gets null
    next();
  };
  handleChange = (editor, data, value) => {
    console.log('handleChange', value);  // gets the field value
  };
  render() {
    return (
      <div className="App">
        <CodeMirror
          onChange={this.handleChange}
          onBeforeChange={this.handleBeforeChange}
        />
      </div>
    );
  }
}
export default App;

```

**Comment by scniro** [comment]
Yikes, it [is always `null` here](urlhttps://github.com/scniro/react-codemirror2/blob/master/src/index.tsx#L639). I admittedly can not recall at this moment *why* this is no, but I believe it's because we don't actully need the value because we're not managing state. Regardless, the inconsistency remains. I'll likely add `value` to the cb since an explicit null isn't all that helpful. Thanks for bringing this up 👍

**Comment by scniro** [comment]
@euZebe Fixed with the [4.0.1](https://github.com/scniro/react-codemirror2/releases/tag/4.0.1) release. Note that `value` in this callback is actually the *old* value since we're getting it *before* a change (in the case of uncontrolled)
